presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.sql.ExpressionUtils.extractConjuncts;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+public class SpatialJoinUtils
+{
+    public static final String ST_CONTAINS = "st_contains";
+    public static final String ST_INTERSECTS = "st_intersects";
+
+    private SpatialJoinUtils()
+    {
+    }
+
+    public static List<FunctionCall> extractSupportedSpatialFunctions(Expression filterExpression)
+    {
+        ImmutableList.Builder<FunctionCall> spatialFunctions = ImmutableList.builder();
+        List<Expression> conjuncts = extractConjuncts(filterExpression);
+        for (Expression expression : conjuncts) {
+            if (expression instanceof FunctionCall) {
+                FunctionCall function = (FunctionCall) expression;
+                if (isSupportedSpatialFunction(function)) {
+                    spatialFunctions.add(function);
+                }
+            }
+        }
+        return spatialFunctions.build();
+    }
+
+    private static boolean isSupportedSpatialFunction(FunctionCall functionCall)
+    {
+        String functionName = functionCall.getName().toString();
+        return functionName.equalsIgnoreCase(ST_CONTAINS) || functionName.equalsIgnoreCase(ST_INTERSECTS);
```

Comment:
Would you explain why? In another comment you said "as you know the spatial function, you know the arguments types, you don't need to analyze here." Wouldn't the same be true here? I know the spatial functions and therefore I know that they have 2 arguments.

Reply:
More a matter of taste probably. You can say "I match st_contains(*, *) because I know what it is" or "I match st_contains because I know what it is (btw it has 2 args)".